Protocol Buffers

syntax = "proto3";
package ecommerce;
// Serviço de Encomendas
service OrderService {
rpc CreateOrder(CreateOrderRequest) returns (CreateOrderResponse);
}
message CreateOrderRequest {
string user_id = 1;
string product_id = 2;
double amount = 3;
}

message CreateOrderResponse {
string order_id = 1;
string status = 2;
}

// Serviço de Pagamentos
service PaymentService {
rpc ProcessPayment(PaymentRequest) returns (PaymentResponse);
}
message PaymentRequest {
string order_id = 1;
double amount = 2;
}
message PaymentResponse {
bool success = 1;
string transaction_id = 2;
}
